tests/Api.Integration.Tests/RegisterTests.cs

```diff
@@ -0,0 +1,203 @@
+using System.Net;
+using System.Net.Http.Json;
+using System.Text.Json;
+using Bogus;
+using Fido2NetLib;
+using FluentAssertions;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.VirtualAuth;
+using Passwordless.Api.Endpoints;
+using Passwordless.Service.Models;
+using Xunit;
+
+namespace Passwordless.Api.Integration.Tests;
+
+public class RegisterTests : IClassFixture<PasswordlessApiFactory>
+{
+    private readonly HttpClient _client;
+
+    private static readonly Faker<RegisterToken> TokenGenerator = new Faker<RegisterToken>()
+        .RuleFor(x => x.UserId, Guid.NewGuid().ToString())
+        .RuleFor(x => x.DisplayName, x => x.Person.FullName)
+        .RuleFor(x => x.Username, x => x.Person.Email)
+        .RuleFor(x => x.Attestation, "None")
+        .RuleFor(x => x.Discoverable, true)
+        .RuleFor(x => x.UserVerification, "Preferred")
+        .RuleFor(x => x.Aliases, x => new HashSet<string> { x.Person.FirstName })
+        .RuleFor(x => x.AliasHashing, false)
+        .RuleFor(x => x.ExpiresAt, DateTime.UtcNow.AddDays(1))
+        .RuleFor(x => x.TokenId, Guid.Empty);
+
+    public RegisterTests(PasswordlessApiFactory apiFactory)
+    {
+        _client = apiFactory.CreateClient();
+        _client.DefaultRequestHeaders.Add("User-Agent", "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36");
+    }
+
+    [Fact]
+    public async Task Server_Can_Successfully_Retrieve_Token_For_Registration()
```

Comment:
Need to include this `xunit.runner.json`: https://github.com/passwordless/passwordless-dotnet/blob/24661e9bbc064b6ecfae8de7a31393655517d170/tests/Passwordless.Tests/xunit.runner.json#L3-L4

And also no need to capitalize every word. Also consider this naming: `I_can_retrieve_registration_token`.

Here's how an analogous test looks in `Passwordless.NET`: https://github.com/passwordless/passwordless-dotnet/blob/24661e9bbc064b6ecfae8de7a31393655517d170/tests/Passwordless.Tests/TokenTests.cs#L16-L30